common/auth/federation_client.go

```diff
@@ -32,16 +33,19 @@ type x509FederationClient struct {
 	securityToken                     securityToken
 	authClient                        *common.BaseClient
 	mux                               sync.Mutex
+	refresh                           bool
 }
 
-func newX509FederationClient(region common.Region, tenancyID string, leafCertificateRetriever x509CertificateRetriever, intermediateCertificateRetrievers []x509CertificateRetriever) federationClient {
+func newX509FederationClient(region common.Region, tenancyID string, leafCertificateRetriever x509CertificateRetriever, intermediateCertificateRetrievers []x509CertificateRetriever, refresh bool) federationClient {
```

Comment:
I'm not a big fan of boolean parameters to configure behaviors of an object.  We might want to provide separate factory methods for each case instead, such as "newRefreshableX509FederationClient()" or something like that.  (Excuse me, I cannot come up with nice names. :P )

Reply:
Even better we might be able to remove that logic entirely. As I read it, it's just checking that the certificate tenancy ID matches that set for the federation client. If they don't match I assume the certificate would be invalid and fail anyway so removing that logic might be an easier option? Unsure what the purpose is.

Reply:
I hope this comment is re-considered.  This article for example describes in more details: https://martinfowler.com/bliki/FlagArgument.html